docker_images/sklearn/app/pipelines/common.py

```diff
@@ -37,21 +38,29 @@ def __init__(self, model_id: str):
                 config = json.load(f)
         except Exception:
             config = dict()
-            raise UserWarning("`config.json` does not exist or is invalid.")
+            warnings.warn("`config.json` does not exist or is invalid.")
 
         self.model_file = (
             config.get("sklearn", {}).get("model", {}).get("file", DEFAULT_FILENAME)
         )
+        self.model_format = config.get("sklearn", {}).get("model_format", "pickle")
 
         try:
             with warnings.catch_warnings(record=True) as record:
-                self.model = joblib.load(
-                    open(Path(cached_folder) / self.model_file, "rb")
-                )
-                if len(record) > 0:
-                    # if there's a warning while loading the model, we save it so
-                    # that it can be raised to the user when __call__ is called.
-                    self._load_warnings += record
+                if self.model_format == "pickle":
+
+                    self.model = joblib.load(
+                        open(Path(cached_folder) / self.model_file, "rb")
+                    )
+                    if len(record) > 0:
```

Comment:
Should this whole check not be outside of the `if self.model_format == "pickle":` condition? Otherwise, the warning is only recorded when the format is pickle.